examples/05_assets/main.rs

```diff
@@ -4,60 +4,51 @@
 extern crate amethyst;
 extern crate cgmath;
 
-use amethyst::{Application, Event, State, Trans, VirtualKeyCode, WindowEvent};
-use amethyst::asset_manager::{AssetLoader, AssetLoaderRaw, AssetManager, Assets, DirectoryStore};
+use amethyst::{Application, Engine, Event, State, Trans, VirtualKeyCode, WindowEvent};
+use amethyst::asset_manager::{AssetFormat, AssetLoader, DirectoryStore, Import, ImportError};
 use amethyst::config::Element;
-use amethyst::ecs::World;
-use amethyst::ecs::components::{LocalTransform, Mesh, Texture, Transform};
+use amethyst::ecs::components::{LocalTransform, Renderable, Texture, Transform};
 use amethyst::ecs::resources::{Camera, Projection, ScreenDimensions};
 use amethyst::gfx_device::DisplayConfig;
-use amethyst::renderer::{Layer, PointLight, Pipeline, VertexPosNormal};
+use amethyst::renderer::{Layer, PointLight, VertexPosNormal};
 use amethyst::renderer::pass::{Clear, DrawShaded};
 use cgmath::{Deg, Euler, Quaternion};
-use std::env::set_var;
 use std::str;
 
 // Implement custom asset loader that reads files with a simple format of
 // 1 vertex and 1 normal per line, with coordinates separated by whitespace.
-struct CustomObj {
-    vertices: Vec<[f32; 3]>,
-    normals: Vec<[f32; 3]>,
+struct CustomObj;
+
+impl AssetFormat for CustomObj {
+    fn file_extensions(&self) -> &[&str] {
```

Comment:
Do we really want to export the macro? I just thought I'd leave it internal. Oh I found another way to return a `&'static [&'static str]`:

```rust
const EXTENSIONS: &'static [&'static str] = &["jpg"];

EXTENSIONS
```

Would this one be okay or should I still export it?